### DISCUSSION TITLE
I want to contribute this "php artisan make:service ExampleService"

### DISCUSSION THREAD

**Discussion by sohag-pro** [laravel/framework#44048] [Ideas] [answered]
Hi,

I want to contribute in Laravel with this *make:service* command. 
It'll create a file in app/Services directory. Where the business logics and utilities can be written. 

Is there anyone working on something like this? 
Or is it a valid feature to work on? 

@taylorotwell
@rodrigopedra 
@nunomaduro

**Comment by rodrigopedra** [comment, marked as answer]
Hey @sohag-pro , while I am not a core team member, I am not aware of anyone working on something like this.

On the other hand, Laravel doesn't dictate how a dev should structure their app regarding business logic. The current factory commands are convenience factories for common framework components, but in reality you could place then wherever you want - although keeping the conventions helps a lot on a long lived project, specially when onboarding new devs with prior Laravel experience.

This way I don't see how having a dedicated command to build services classes would become part of the core, as not being a core component, there are many ways one can decide to structure their project's business logic. Some people like actions classes (as seen in laravel/fortify), other like dedicated query commands, or defer most of their business logic to jobs (queued or sync).

But if you think this is something you believe should be part of the core, give it a try. But in my opinion such factory command is more suitable to a third-party package.

Have a nice day =)

**Reply by henzeb** [reply]
I agree, and then there are a few of them already doing things like this. For example: https://github.com/theanik/laravel-more-command

**Reply by sohag-pro** [reply]
@rodrigopedra thank you for your time and effort you gave for the answer. I'll give it a try.

**Reply by rodrigopedra** [reply]
You're welcome! Have a nice day, and good luck =)
